rustls-cert-gen/src/cert/entity.rs

```diff
@@ -0,0 +1,127 @@
+use rcgen::{
+	Certificate, CertificateParams, DnType, ExtendedKeyUsagePurpose, IsCa, KeyUsagePurpose, SanType,
+};
+
+use super::{Cert, CertParams, SerializedEntity};
+/// End-entity [Certificate]
+pub struct EndEntity {
+	cert: Certificate,
+}
+
+impl Cert for EndEntity {
+	/// Sign with `self.signer` and serialize.
+	fn serialize(&self, signer: Option<&Certificate>) -> Result<SerializedEntity, rcgen::Error> {
+		let cert_pem = self.cert.serialize_pem_with_signer(signer.unwrap())?;
+		let key_pem = self.cert.serialize_private_key_pem();
+		Ok(SerializedEntity { cert_pem, key_pem })
+	}
+	fn cert(&self) -> &Certificate {
+		&self.cert
+	}
+}
+
+impl CertParams for EndEntityParams {
+	fn params(&self) -> &CertificateParams {
+		&self.params
+	}
+}
+
+/// [CertificateParams] from which an [EndEntity] [Certificate] can be built
+pub struct EndEntityParams {
+	params: CertificateParams,
+}
+
+impl EndEntityParams {
+	/// Initialize `EndEntityParams`
+	pub fn new(mut params: CertificateParams) -> Self {
+		params.is_ca = IsCa::NoCa;
+		params.use_authority_key_identifier_extension = true;
+		params.key_usages.push(KeyUsagePurpose::DigitalSignature);
+		Self { params }
+	}
+	pub fn cn(mut self, name: &str) -> Self {
```

Comment:
I think we should avoid using acronyms in the API if possible
```suggestion
	pub fn common_name(mut self, name: &str) -> Self {
```